source/extensions/filters/network/thrift_proxy/router/router_impl.cc

```diff
@@ -48,6 +48,37 @@ RouteEntryImplBase::RouteEntryImplBase(
   }
 }
 
+// Similar validation procedure with Envoy::Router::RouteEntryImplBase::validateCluster
+void RouteEntryImplBase::validateClusters(
+    const Upstream::ClusterManager::ClusterInfoMaps& cluster_info_maps) const {
+  // Currently, we verify that the cluster exists in the CM if we have an explicit cluster or
+  // weighted cluster rule. We obviously do not verify a cluster_header rule. This means that
+  // trying to use all CDS clusters with a static route table will not work. In the upcoming RDS
+  // change we will make it so that dynamically loaded route tables do *not* perform CM checks.
+  // In the future we might decide to also have a config option that turns off checks for static
+  // route tables. This would enable the all CDS with static route table case.
+  if (!cluster_name_.empty()) {
```

Comment:
Isn't this the same logic as in `source/common/router/conifg_impl.cc`?

Reply:
Yes, as line 51 said, it's similar validation procedure with `Envoy::Router::RouteEntryImplBase::validateCluster`
However, they are not with the same interface (as I need to adjust some method name). I have no idea how much effort reusing the logic is.